### ISSUE TITLE
Static content in tab order - failure of 2.4.3 or not?

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#4634]
A discussion that seems to keep bubbling up (most recently on WebAIM https://webaim.org/discussion/mail_thread?thread=10037) is whether or not static content that receives focus (e.g. because it has been marked with `tabindex="0"`) is a failure of 2.4.3 or not.

My take for the longest time has been that, unless the *order* that these static elements receive focus in makes stuff illogical/affects meaning of the overall content, that this is fine. I don't think static content receiving focus violates the normative wording of the SC

> If a Web page can be navigated sequentially and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability.

I also don't see anything in the understanding document that suggests anything about static content being focusable as being a failure.

To be clear, I would generally flag inappropriately focusable static content as something that a developer should address, but I make the point of clarifying that it doesn't strictly (to me at least) fail 2.4.3 (or any other SC). That it's more of a usability problem.

Long story short: it would be good to get a consensus position on this, and to add an example/clarification/note to the understanding document about this.

**Comment by patrickhlauke** [comment]
@kfranqueiro @mbgower it seems this issue was transferred in January to the `wcag3` repo, but it's still an issue that relates to WCAG 2.x. suggest it gets transferred back ... and if it was added to 3 purely because of "this is something we'll want to tackle in wcag3", then a new separate fresh issue is opened in `wcag3` about it. /cc @cwadamsoforacle

**Comment by alastc** [comment]
Yep, sorry, we were a bit liberal with moving wcag.next items to WCAG3, we can put this back.

**Comment by philljenkins** [comment]
Reviewing the file in the [PR #1643](https://github.com/w3c/wcag/pull/1643/files)
> - Static/non-interactive elements can receive focus, as long as they don't impede operation of the content, or result in confusing or illogical focus order.

I think this bullet is still problematic.  The tabbing operation is already considered laborious by many. Tab, Tab, Tab ... ugh:
1. Doesn't a single static/non-interactive "heading" that receives focus impede operation? If not a single, would two unnecessary stops? Or is a single unnecessary Tab stop being implied as meeting the definition of "impede" and therefore fails the SC? I vote Yes. [However, that should be made more explicit in the Understanding page.]
2. A single extra unnecessary Tab stop also meets the definition of "confusing or illogical".

You could argue that it is a **_temporary_** impediment or confusion. But temporary (as defined by what/whom?) is still something that was not expected, and that is the essence of the SC in my opinion. 

### Why static elements in the Tab order are usually a problem
- **_Impede operation:_** Every extra tab stop adds unnecessary keystrokes, which can significantly slow down users who rely on a keyboard interface for navigation, such as those with motor disabilities. A sighted keyboard user expects to tab only through interactive controls like links, buttons, and form fields.
- **_Confusing and illogical_**: When a static element like a heading receives focus, a sighted user might see a focus indicator appear on something they cannot interact with, which is confusing. For a screen reader user, this can be even more disorienting. Screen readers announce elements in the tab order, so unexpectedly encountering a non-functional element can create confusion about the page's structure.
- **_False affordance_**: Making a non-interactive element focusable can give a user the false impression that it is interactive. They may press a key expecting an action to occur, only to be met with no response, which is a frustrating experience. 

The relationship to [4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value) Failure Technique is also important:
- [Failure of Success Criterion 4.1.2 due to using script to make div or span a user interface control in HTML without providing a role for the control](https://www.w3.org/WAI/WCAG21/Techniques/failures/F59.html)
- However, my concern is the flip of F59.  My concern is the Tab stopping on something that isn't and **_shouldn't_** be operable, while F59 is about an element that should be operable, has a Tab stop, but does not have the correct `role` defined. 

Finally, it has also been a few years since this issue was opened and comments made.  For example, both Chrome and Firefox now automatically provide Tab stop to elements that are/become scrollable without requiring them to have interactive elements within or Tab stop added by the author.

**Comment by philljenkins** [comment]
### The exception to the rule
The "as long as they don't impede operation" clause is not a blanket permission for making static content focusable. It is meant to cover rare and highly specific edge cases where doing so genuinely helps the user. Examples include: 
- **_Specialized application interfaces_**: Some applications, particularly single-page apps (SPAs), may need to set programmatic focus on a heading or region to orient users after a new section of content is loaded.
- **_Alert or dialog boxes_**: It can sometimes be necessary to move focus to a heading or message in a newly opened dialog to ensure a screen reader announces the context.
- **_Container elements_**: In complex widgets like a tab panel, it may be necessary to place focus on a container or heading that is part of the control group

**Comment by patrickhlauke** [comment]
>  The tabbing operation is already considered laborious by many. Tab, Tab, Tab ... ugh

well, we're not going to be able to tackle that...

> Doesn't a single static/non-interactive "heading" that receives focus impede operation?

that sounds like an extreme interpretation. you're claiming that just a single extra tab stop makes a site/web app go from "accessible" to "not accessible" (and therefore, illegal in some legislative frameworks)? please...

**Comment by philljenkins** [comment]
> ... you're claiming that just a single extra tab stop makes a site/web app go from "accessible" to "not accessible" (and therefore, illegal in some legislative frameworks)? please...

No, I'm claiming that the SC needs better understanding documentation.  Otherwise, if/when legislative framework(s) are strictly interpreting SC failures, there needs to be more examples in the understanding documentation of when it fails and when it doesn't.

**Comment by TestPartners** [comment]
In previous discussions, there has been a strong consensus that "operable" means "can be operated", not "is easy to operate". If we accept this, no further explanation is needed. If we don't accept it, we're into a whole world of subjective judgements that are better left to WCAG3.

**Comment by philljenkins** [comment]
@TestPartners are you suggesting tweaking the existing bullet in the file in the [PR #1643](https://github.com/w3c/wcag/pull/1643/files) by adding the [easy] term is removing attention to "any" impediment?
or, make the term [operation] defined?

> -Static/non-interactive elements can receive focus, as long as they don't impede [easy] operation of the content, or result in confusing or illogical focus order.

**Comment by GreggVan** [comment]
@TestPartners 

You provide a choice of just two options.   However neither one is the one used by WCAG.

To base it on an evaluation of operable (at any level  easy to minimum)  -- we would have to say operable by who?   An able bodied person?   A person with X degree of Y disability?

Rather than say something is operable we just say "that for something to be accessible to anyone they must be able to perceive, operate, and understand it - including if they depend on AT  (or not) to do this or if they have a disability nor not. 

Then we lay out general guidelines and specific objective measures as SC for MINIMUM accessibility for different aspects of  P O U and R.    

 Each of these is meant to be a bright line (objective) or as bright a line as we can.  

*important*\
whenever you stand on that line - you can distinguish no difference in accessibility from just on one side and just on the other.   And you could move that line a tad and you would have a new "minimum" value for accessibility that would be no different than what you had before.    The line is arbitrary and will ALWAYS  divide people who can or can't or can more easily or not into two camps depending on which side of the line they fall.    
- This is not great - but it is the best we can do.  The line has to be drawn someplace. 
- the line is dawn using a consensus basis of the working group with members from industry, consumers, academia, etc. etc. 
- BEST PRACTICE is to always design  products as far (to the good side) as you can AWAY from that line -- including as many people as you can.
- but if you are just barely on the good side of the line - you pass
- if you are just barely on the bad side of the line - you  fail
- if you want to avoid arguments - don't get near the line
- if you are evaluating - do not argue with the line or where it is.   it is where it is
- if you are part of the team drawing the line - AH  There is where we need debate.  but effective and polite debate


My personal Advice for those who are creating guidelines/regulations/standards.

- look at the issues from all sides and present you best argument for where one thinks the line should be. 
-  Don't argue that it will be better for a line to be somewhere else for some people.  
     - That is not an argument -- that is a statement we all agree on.    
- Present an argument based on the constraints of all sides -- and one that one thinks one can get agreement on. 
    -  do not say "i don't like" something.  Every member of the working group doesn't like something.  Most people don't like what they propose even - they just can't think of something that is better.
- it is ok to critique if one has a suggestion for how to solve it.    And if so present it and see if one can get agreement
- it is also ok to grumble  (politely) that one isn't happy but can't see a better solution.  and see if someone else does.
- but grumble against situations or words -- not against people.   Everyone is trying.  Some are better at it than others but I  have not seen a someone who did not care on this group -- even people who somethings are not very polite or political about it.  (But don't be one who is not polite - no matter how much one cares).
- and be open to the fact that it is a balancing act.  
      - listen to the other arguments and see if one can find common ground or a balance point. 

And remember -- this is really hard.  Really hard.

**Comment by patrickhlauke** [comment]
removing this from the WCAG 2.x project, as there is a PR that covers this
